reporters/dogstatsd.go

```diff
@@ -84,3 +102,25 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) scheduleRestart(timeout time.Duration) {
+	if timeout.Nanoseconds() > 0 {
+		time.AfterFunc(timeout, func() {
+			if d.client != nil {
+				if err := d.statsdClient.Close(); err != nil {
+					d.logger.Errorf("DogStatsD: failed to close statsd connection during restart: %s", err.Error())
+				}
+			}
+
+			c, err := dogstatsd.New(d.host, d.prefix)
+			if err == nil {
+				d.statsdClient = c.Client.(*statsd.Client)
+				d.client = c
+			} else {
+				d.logger.Errorf("DogStatsD: failed to recreate dogstatsd client during restart. %s", err.Error())
+				d.client = nil
```

Comment:
```suggestion
```
this can't be null, what about just using the old dogstatsd instance?

Reply:
Oh, yeah! I was in doubt about how to deal with the old client instance. I think it's better to maintain old references indeed.

Reply:
Thanks for the comment!

Reply:
Fixed!